cmd/notification.go

```
@@ -1101,6 +1101,61 @@ func (sys *NotificationSys) GetBandwidthReports(ctx context.Context, buckets ...
 	return consolidatedReport
 }
 
+// GetBucketMetrics - gets the cluster level bucket metrics from all nodes excluding self.
+func (sys *NotificationSys) GetBucketMetrics(ctx context.Context) <-chan Metric {
+	if sys == nil {
+		return nil
+	}
+	g := errgroup.WithNErrs(len(sys.peerClients))
+	peerChannels := make([]<-chan Metric, len(sys.peerClients))
+	for index := range sys.peerClients {
+		if sys.peerClients[index] == nil {
+			continue
```

I think you need to return an error here or at least ignore peersChannels[index] with a nil value before running the next goroutine

---

No this is not returning any error @vadmeste this is simply copy paste from GetClusterMetrics()

---

and we also don't need to return errors for `nil` peerClients since it is always nil and we do not need to log it.